code/code/misc/combat.cc

```diff
@@ -2047,6 +2047,9 @@ static void checkLearnFromHit(TBeing * ch, int tarLevel, TThing * o, bool isPrim
         if (obj && ch->doesKnowSkill(SKILL_2H_SPEC_DEIKHAN) && obj->isPaired()) {
           ch->learnFromDoingUnusual(LEARN_UNUSUAL_NORM_LEARN, SKILL_2H_SPEC_DEIKHAN, max(0, (100 - (2* myLevel))));
         }
+        if (obj && ch->doesKnowSkill(SKILL_2H_SPEC) && obj->isPaired()) {
```

Comment:
Same. Do we need to call get skill num or is that included in doesknowskill()?

Reply:
This still needs fixing

Reply:
Just another comment to point out that the fix here is the same for the other places I mentioned - replacing SKILL_2H_SPEC with `getSkillNum(SKILL_2H_SPEC)`.